apps/previewgenerator.sh

```diff
@@ -16,67 +16,258 @@ debug_mode
+    do
+        case $choice in
+            "PNG")
+                occ_command config:system:set enabledPreviewProviders 11 --value="OC\\Preview\\PNG"
+            ;;
+
+            "JPEG")
+                occ_command config:system:set enabledPreviewProviders 12 --value="OC\\Preview\\JPEG"
+            ;;
+
+            "GIF")
+                occ_command config:system:set enabledPreviewProviders 13 --value="OC\\Preview\\GIF"
+            ;;
+
+            "BMP")
+                occ_command config:system:set enabledPreviewProviders 14 --value="OC\\Preview\\BMP"
+            ;;
+
+            "MarkDown")
+                occ_command config:system:set enabledPreviewProviders 15 --value="OC\\Preview\\MarkDown"
+            ;;
+
+            "MP3")
+                occ_command config:system:set enabledPreviewProviders 16 --value="OC\\Preview\\MP3"
+            ;;
+
+            "TXT")
+                occ_command config:system:set enabledPreviewProviders 17 --value="OC\\Preview\\TXT"
+            ;;
+
+            "Movie")
+                occ_command config:system:set enabledPreviewProviders 18 --value="OC\\Preview\\Movie"
+            ;;
+
+            *)
+            ;;
+        esac
+    done 8< results
+    rm -f results
+    clear
+fi
+
+# Set aspect ratio
+occ_command config:app:set previewgenerator squareSizes --value="32 256"
+occ_command config:app:set previewgenerator widthSizes  --value="256 384"
+occ_command config:app:set previewgenerator heightSizes --value="256"
+occ_command config:system:set preview_max_x --value="2048"
+occ_command config:system:set preview_max_y --value="2048"
+occ_command config:system:set jpeg_quality --value="60"
+occ_command config:app:set preview jpeg_quality --value="60"
+
+msg_box "In the last step you can define a specific Nextcloud user for which will be the user that runs the preview-generation. 
+
+The default behavoiur (just hit [ENTER]) is to run with the system user 'www-data' which will generate previews for all users. 
+
+If you on the other hand choose to use a specific user, previews will ONLY be generated for that specific user."
+if [[ "no" == $(ask_yes_or_no "Do you want to choose a specific Nextcloud user to generate previews?") ]]
+then
+    # Add crontab for www-data
+    crontab -u www-data -l | { cat; echo "0 4 * * * php -f $NCPATH/occ preview:pre-generate >> $VMLOGS/previewgenerator.log"; } | crontab -u www-data -
+    touch "$VMLOGS"/previewgenerator.log
+    chown www-data:www-data "$VMLOGS"/previewgenerator.log
+    
     # Pre generate everything
     occ_command preview:generate-all
+else
+    while true
+    do
+    PREVIEW_USER=$(whiptail --inputbox "Enter the Nextcloud user for which you want to run the preview-generation" "$WT_HEIGHT" "$WT_WIDTH" 3>&1 1>&2 2>&3)
+    if ! occ_command user:list | grep "$PREVIEW_USER" | awk '{print $3}'
+        then
+            msg_box "It seems like the user you entered ($PREVIEW_USER) doesn't exist, please try again."
```

Comment:
the check somehow doesn't get triggered. 
do you have another idea how to do it?

Reply:
Yeah, that's one of the bugs. I'll think about it. If you remove `awk print` you get another error.

Reply:
The most "secure" option would be to use `read` as we do in the other scrips.

Reply:
This is fixed.